PR: https://github.com/odoo/odoo/pull/239029

From: 1d94414a0d78f8760f2089d1a7c635eac55942f4
From: lebm-odoo
Date: 2026-02-22 13:39:05

Breaking data model changes scores: del:2 + add:0, change matches:
-    work_entry_source_calendar_invalid = fields.Boolean(related="version_id.work_entry_source_calendar_invalid", inherited=True, groups="hr.group_hr_manager")
-    work_entry_source_calendar_invalid = fields.Boolean(

Total Changes: 33

[IMP] hr_work_entry: work_entry_source becomes "Tracking Method"

New setting is added to set a default tracking method.

task-5362147

closes odoo/odoo#239029

Related: odoo/enterprise#101565
Related: odoo/upgrade#9032
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_work_entry/models/__init__.py
+++ b/addons/hr_work_entry/models/__init__.py
@@ -1,6 +1,6 @@
-# -*- coding:utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from . import res_company
 from . import hr_work_entry_type
 from . import hr_user_work_entry_employee
 from . import resource_calendar
@@ -8,5 +8,4 @@ from . import resource_calendar_attendance
 from . import resource_calendar_leaves
 from . import hr_employee
 from . import hr_version
-from . import res_company
 from . import res_config_settings

--- a/addons/hr_work_entry/models/hr_employee.py
+++ b/addons/hr_work_entry/models/hr_employee.py
@@ -7,7 +7,6 @@ class HrEmployee(models.Model):
     _inherit = 'hr.employee'
 
     work_entry_source = fields.Selection(readonly=False, related="version_id.work_entry_source", inherited=True, groups="base.group_system,hr.group_hr_manager")
-    work_entry_source_calendar_invalid = fields.Boolean(related="version_id.work_entry_source_calendar_invalid", inherited=True, groups="hr.group_hr_manager")
     external_code = fields.Char("External Code", copy=False, help="Use this code to export your data to a third party", groups="hr.group_hr_user")
 
     # YTI TODO: Rename private method into _get_work_entries_vals()

--- a/addons/hr_work_entry/models/hr_version.py
+++ b/addons/hr_work_entry/models/hr_version.py
@@ -15,22 +15,14 @@ from odoo.tools.intervals import Intervals
 class HrVersion(models.Model):
     _inherit = 'hr.version'
 
-    work_entry_source = fields.Selection([('calendar', 'Working Schedule')], required=True, default='calendar', tracking=True, help='''
-        Defines the source for work entries generation
-
-        Working Schedule: Work entries will be generated from the working hours below.
-        Attendances: Work entries will be generated from the employee's attendances. (requires Attendance app)
-        Planning: Work entries will be generated from the employee's planning. (requires Planning app)
-    ''', groups="base.group_system,hr.group_hr_manager")
-    work_entry_source_calendar_invalid = fields.Boolean(
-        compute='_compute_work_entry_source_calendar_invalid',
-        groups="hr.group_hr_manager",
-    )
-
-    @api.depends('work_entry_source', 'resource_calendar_id')
-    def _compute_work_entry_source_calendar_invalid(self):
-        for version in self:
-            version.work_entry_source_calendar_invalid = version.work_entry_source == 'calendar' and not version.resource_calendar_id
+    work_entry_source = fields.Selection(string="Tracking Method", selection=[
+        ('calendar', 'Time Off'),
+    ], default=lambda self: self.env.company.sudo().work_entry_source, tracking=True, required=True, help='''
+            Defines the source for work entries generation
+
+            Working Schedule: Work entries will be generated from the working hours below.
+            Attendances: Work entries will be generated from the employee's attendances. (requires Attendance app)
+        ''', groups="base.group_system,hr.group_hr_manager")
 
     # YTI Break ormcache + select country attendance entry type
     @ormcache()

--- a/addons/hr_work_entry/models/res_company.py
+++ b/addons/hr_work_entry/models/res_company.py
@@ -7,3 +7,6 @@ class ResCompany(models.Model):
     _inherit = 'res.company'
 
     external_code = fields.Char("External Code", groups="hr.group_hr_user", copy=False, help="Use this code to export your data to a third party")
+    work_entry_source = fields.Selection(string="Tracking Method", selection=[
+        ('calendar', 'Time'),
+    ], default='calendar', required=True, groups="hr.group_hr_user")

--- a/addons/hr_work_entry/models/res_config_settings.py
+++ b/addons/hr_work_entry/models/res_config_settings.py
@@ -8,3 +8,5 @@ class ResConfigSettings(models.TransientModel):
 
     external_code = fields.Char(
         related='company_id.external_code', string="External Code", readonly=False)
+    work_entry_source = fields.Selection(string='Default Tracking', related='company_id.work_entry_source',
+                                         groups="hr.group_hr_user", readonly=False)
